src/confirmation-popover.directive.ts

```diff
@@ -236,7 +241,8 @@ export class ConfirmationPopoverDirective
     if (
       this.popover &&
       !this.elm.nativeElement.contains(event.target) &&
-      !this.popover.location.nativeElement.contains(event.target)
+      !this.popover.location.nativeElement.contains(event.target) &&
+      this.closeOnOutsideClick
```

Comment:
Change this to `typeof this.closeOnOutsideClick !== 'undefined' ? this.closeOnOutsideClick : this.defaultOptions.closeOnOutsideClick` to use the default value if set